toolkit/analysis_and_visualization/classification/generate-visualization.py

```diff
@@ -59,2520 +79,887 @@ def readHierarchyFile(hierarchyFile):
-    print("");
-    print('Total Top K match -- '+str(passCount));
-    accuracyPer = float(passCount);
-    accuracyPer = (accuracyPer / netSummaryImages) * 100;
-    print('Inference Accuracy on Top K -- '+str(np.around(accuracyPer,decimals=2))+' %');
-    print('Average Pass Probability for Top K -- '+str(np.around(avgPassProb,decimals=4)));
-    print("");    
-    print('Total mismatch -- '+str(totalMismatch));
-    accuracyPer = float(totalMismatch);
-    accuracyPer = (accuracyPer/netSummaryImages) * 100;
-    print('Inference mismatch Percentage -- '+str(np.around(accuracyPer,decimals=2))+' %');
+
+def copyHtmlAssets(outputDirectory, fileName):
+    """Makes required output directories and copies all required html files.
+    Copies icons, tablesort file inside util directory, all scripts, styles and 
+    main html file  inside the output directory
+
+    Args:
+        outputDirectory (string): root directory path to copy files to
+        fileName (string): name of output directory where the files will be copied to
+
+    Returns:
+        tuple: (created output directory path, results directory path)
+    """
+    # create toolkit with icons and images
+    toolKit_Dir = os.path.join(outputDirectory, fileName + '-toolKit')
+    toolKit_dir = os.path.expanduser(toolKit_Dir)
+    if not os.path.exists(toolKit_dir):
+        os.makedirs(toolKit_dir)
+    # copy scripts and icons
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    fromDirectory = os.path.join(dir_path, 'icons')
+    toDirectory = os.path.join(toolKit_dir, 'icons')
+    copy_tree(fromDirectory, toDirectory)
+
+    new_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    fromDirectory = os.path.join(new_path, 'utils')
+    toDirectory = os.path.join(toolKit_dir, 'utils')
+    copy_tree(fromDirectory, toDirectory)
+
+    resultsDirectory = os.path.join(toolKit_dir, 'results')
+    if not os.path.exists(resultsDirectory):
+        os.makedirs(resultsDirectory)
+
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+
+    dirsToCopy = ['scripts', 'styles']
+    for srcDir in dirsToCopy:
+        fromDirectory = os.path.join(dir_path, 'assets', srcDir)
+        toDirectory = os.path.join(toolKit_Dir, srcDir)
+        copy_tree(fromDirectory, toDirectory)
+
+    # COpy the main template files
+    shutil.copy(os.path.join(dir_path, 'assets', 'templates',
+                             'index.html'), os.path.join(toolKit_Dir, 'index.html'))
+
+    return toolKit_dir, resultsDirectory
+
+
+def copyImages(imagesSource, toolkit_dir):
+
+    dest = os.path.join(toolkit_dir, 'images')
+    copy_tree(imagesSource, dest)
+    return dest
+
+
+def generateTop1Result(resultsDirectory, resultDataBase, labelLines):
+    logger.debug("Generating top 1 result file")
+    outputCsvFile = os.path.join(resultsDirectory, 'results.csv')
+    outputJsonFile = os.path.join(resultsDirectory, 'results.json')
```

Comment:
Codacy found an issue: [local variable 'outputJsonFile' is assigned to but never used (F841)](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6397592)